Use NetworkTransportStream to represent all WebTransport streams in NetworkProcess

AnkshitJain opened the pull request:
#### 640a8a4349a937a95cbe1c8f02df6366707d3e9a
<pre>
Use NetworkTransportStream to represent all WebTransport streams in NetworkProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=284014">https://bugs.webkit.org/show_bug.cgi?id=284014</a>
<a href="https://rdar.apple.com/140890615">rdar://140890615</a>

Reviewed by Alex Christensen.

NetworkTransportBidirectionalStream, NetworkTransportSendStream, and NetworkTransportReceiveStream are used to define different WebTransport stream types. This is better handled using a single NetworkTransportStream type to handle the underlying WebTransport stream on the network.

The streams share two operations, read and write, where the stream type decides which operations are permitted. This abstraction allows the network to be hooked to a single stream type, reducing the duplication of read and write operations across 3 stream types.

* Source/WebKit/NetworkProcess/webtransport/NetworkTransportBidirectionalStream.cpp: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportBidirectionalStream.h: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSendStream.cpp: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSendStream.h: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSession.cpp:
(WebKit::NetworkTransportSession::sendStreamSendBytes):
(WebKit::NetworkTransportSession::streamSendBytes):
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSession.h:
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportStream.cpp: Renamed from Source/WebKit/NetworkProcess/webtransport/NetworkTransportReceiveStream.cpp.
(WebKit::NetworkTransportStream::NetworkTransportStream):
(WebKit::NetworkTransportStream::sendBytes):
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportStream.h: Renamed from Source/WebKit/NetworkProcess/webtransport/NetworkTransportReceiveStream.h.
(WebKit::NetworkTransportStream::create):
(WebKit::NetworkTransportStream::identifier const):
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportBidirectionalStreamCocoa.mm: Removed.
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSendStreamCocoa.mm: Removed.
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm:
(WebKit::NetworkTransportSession::createBidirectionalStream):
(WebKit::NetworkTransportSession::createOutgoingUnidirectionalStream):
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportStreamCocoa.mm: Renamed from Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportReceiveStreamCocoa.mm.
(WebKit::NetworkTransportStream::NetworkTransportStream):
(WebKit::NetworkTransportStream::sendBytes):
(WebKit::NetworkTransportStream::receiveLoop):
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/287358@main">https://commits.webkit.org/287358@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ddc3ce26b78ef646e13a65d1c2fd6a2d55d485b4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79342 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58370 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32719 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83956 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/30490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/67444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6631 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/62062 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/30490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82409 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/67444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/72290 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/42370 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/67444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28888 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/67444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/26921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85356 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6625 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/4613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/70311 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/6789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/68166 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/69558 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/12472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12245 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/6577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/12323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/6496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/9969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/8292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportStreamCocoa.mm:
```diff
@@ -24,25 +24,40 @@
-void NetworkTransportReceiveStream::receiveLoop()
+void NetworkTransportStream::sendBytes(std::span<const uint8_t> data, bool)
 {
+    if (m_streamType == NetworkTransportStreamType::IncomingUnidirectional)
```
Let's change this to ASSERT(m_streamType != NetworkTransportStreamType::IncomingUnidirectional) like NetworkTransportStream::receiveLoop

Commit:
Use NetworkTransportStream to represent all WebTransport streams in NetworkProcess
https://bugs.webkit.org/show_bug.cgi?id=284014
rdar://140890615

Reviewed by Alex Christensen.

NetworkTransportBidirectionalStream, NetworkTransportSendStream, and NetworkTransportReceiveStream are used to define different WebTransport stream types. This is better handled using a single NetworkTransportStream type to handle the underlying WebTransport stream on the network.

The streams share two operations, read and write, where the stream type decides which operations are permitted. This abstraction allows the network to be hooked to a single stream type, reducing the duplication of read and write operations across 3 stream types.

* Source/WebKit/NetworkProcess/webtransport/NetworkTransportBidirectionalStream.cpp: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportBidirectionalStream.h: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSendStream.cpp: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSendStream.h: Removed.
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSession.cpp:
(WebKit::NetworkTransportSession::sendStreamSendBytes):
(WebKit::NetworkTransportSession::streamSendBytes):
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportSession.h:
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportStream.cpp: Renamed from Source/WebKit/NetworkProcess/webtransport/NetworkTransportReceiveStream.cpp.
(WebKit::NetworkTransportStream::NetworkTransportStream):
(WebKit::NetworkTransportStream::sendBytes):
* Source/WebKit/NetworkProcess/webtransport/NetworkTransportStream.h: Renamed from Source/WebKit/NetworkProcess/webtransport/NetworkTransportReceiveStream.h.
(WebKit::NetworkTransportStream::create):
(WebKit::NetworkTransportStream::identifier const):
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportBidirectionalStreamCocoa.mm: Removed.
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSendStreamCocoa.mm: Removed.
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm:
(WebKit::NetworkTransportSession::createBidirectionalStream):
(WebKit::NetworkTransportSession::createOutgoingUnidirectionalStream):
* Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportStreamCocoa.mm: Renamed from Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportReceiveStreamCocoa.mm.
(WebKit::NetworkTransportStream::NetworkTransportStream):
(WebKit::NetworkTransportStream::sendBytes):
(WebKit::NetworkTransportStream::receiveLoop):
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/287358@main
